tools/doimage/doimage.c

```diff
@@ -1578,9 +1578,9 @@ int write_boot_image(uint8_t *buf, uint32_t image_size, FILE *out_fd)
 
 int main(int argc, char *argv[])
 {
-	char in_file[MAX_FILENAME+1];
-	char out_file[MAX_FILENAME+1];
-	char ext_file[MAX_FILENAME+1];
+	char in_file[MAX_FILENAME+1] = { 0 };
+	char out_file[MAX_FILENAME+1] = { 0 };
+	char ext_file[MAX_FILENAME+1] = { 0 };
```

Comment:
I don't think this is a correct change. 

This code is using snprintf in the line 1598, and the only problem that I could say is that it doesn't check the value returned by snprintf, but as MAX_FILENAME is defined to be 255, then it is impossible to have a truncation (even in the case getpid returns a 64 bit value). I wouldn't see a bad thing to add that test even in the case that we know it is impossible to have a truncation.

It  uses strncpy in the line 1618, 1636, 1663 and 1668,  ant that code is wrong even with your patch. The correct solution is to check that the lenght of the parameter is smaller than MAX_FILENAME, and after that you can use memcpy because you know the size. Using something like strlcpy doesn't add anything in this case, you still have to check the returned value, because you can have a string '\0' terminated but truncated, that can generate other security problems.

Reply:
I agree with your last statement. The code should check the actual length of optagr, make sure it's smaller than the copy buffer and only then copy the optagr string to the destination.